api/server/sdk/utils.go

```diff
@@ -86,9 +86,10 @@ func timeWeekdayToSdkWeekly(t time.Weekday) api.SdkTimeWeekday {
 
 func sdkSchedToRetainInternalSpec(
 	req *api.SdkSchedulePolicyInterval,
+	checkRetain bool,
 ) (*sched.RetainIntervalSpec, error) {
 
-	if req.GetRetain() < 1 {
+	if checkRetain && req.GetRetain() < 1 {
```

Comment:
instead of adding a boolean to the function and checking retain here, let's move the check of it to the caller. This will simplify the arguments, and make the caller responsible for checking the retain value.

Reply:
you mean move the check to create/update function of shed_policy.go or leave it unto the client( totally remove the check)?

Reply:
Done. I moved it to cloudbjckup-schedule.go.